packages/browser/src/core/user/index.ts

```diff
@@ -183,18 +200,92 @@ export interface CookieOptions {
   sameSite?: string
 }
 
+export class UniversalStorage {
+  private stores: Store[]
+
+  constructor(stores?: Store[]) {
+    this.stores = stores || []
+  }
+
+  private getStores(storeTypes: StoreType[] | undefined): Store[] {
+    return storeTypes
+      ? this.stores.filter((s) => storeTypes.indexOf(s.getType()) !== -1)
+      : this.stores
+  }
+
+  public push(store: Store) {
+    this.stores.push(store)
+  }
+
+  public getAndSync<T>(key: string, storeTypes?: StoreType[]): T | null {
+    const val = this.get(key, storeTypes)
+
+    return this.set(
+      key,
+      typeof val === 'number' ? val.toString() : val,
```

Comment:
What's the reasoning behind coercing to number here rather than deferring to set? Just flagging this, as it feels like this might be an unintended side effect?

Reply:
This is a legacy behavior. I think the reason seems to be AJS classic may set number cookies, but in AJS next we want to convert those to string ( not sure why, maybe because our cookie library doesn't support them 🤔 ) ... I think the problem with this code shows even more with stronger typings that I just added, hence had to add the `@ts-ignore`. It breaks the assumption that a given cookie key will always have a certain value type :(

Reply:
Gotcha, I kind of missed that this logic already existed -- all of this context would be great as a comment!